src/Features/CSharp/Portable/Snippets/CSharpForLoopSnippetProvider.cs

```diff
@@ -0,0 +1,179 @@
+            if (options != null)
+            {
+                varBuiltInType = options.GetOption(CSharpCodeStyleOptions.VarForBuiltInTypes).Value;
+            }
+
+            var indexVariable = generator.Identifier("i");
+            var varIdentifier = SyntaxFactory.IdentifierName("var");
+
+            // Creating the variable declaration based on if the user has
+            // 'var for built in types' set in their editorconfig.
+            var variableDeclarationSyntax = varBuiltInType
+                ? SyntaxFactory.VariableDeclaration(varIdentifier,
+                    SyntaxFactory.SingletonSeparatedList(
+                        SyntaxFactory.VariableDeclarator(
+                            indexVariable,
+                            argumentList: null,
+                            SyntaxFactory.EqualsValueClause((ExpressionSyntax)generator.LiteralExpression(0)))))
+                : SyntaxFactory.VariableDeclaration(compilation.GetSpecialType(SpecialType.System_Int32).GenerateTypeSyntax(allowVar: false),
+                    SyntaxFactory.SingletonSeparatedList(
+                        SyntaxFactory.VariableDeclarator(
+                            indexVariable,
+                            argumentList: null,
+                            SyntaxFactory.EqualsValueClause((ExpressionSyntax)generator.LiteralExpression(0)))));
+
+            var forLoopSyntax = SyntaxFactory.ForStatement(variableDeclarationSyntax,
+                SyntaxFactory.SeparatedList<ExpressionSyntax>(),
+                (ExpressionSyntax)generator.LessThanExpression(
+                    generator.IdentifierName(indexVariable),
+                    // Using a temporary identifier name for now, could later be changed
+                    // to look for an iterable item in the scope of the insertion.
+                    generator.IdentifierName("length")),
+                SyntaxFactory.SingletonSeparatedList<ExpressionSyntax>(
+                    SyntaxFactory.PostfixUnaryExpression(
+                        SyntaxKind.PostIncrementExpression, SyntaxFactory.IdentifierName(indexVariable))),
+                SyntaxFactory.Block());
+
+            return forLoopSyntax;
+        }
+
+        /// <summary>
+        /// Goes through each piece of the for statement and extracts the identifiers
+        /// as well as their locations to create SnippetPlaceholder's of each.
+        /// </summary>
+        protected override ImmutableArray<SnippetPlaceholder> GetForLoopSnippetPlaceholders(SyntaxNode node, ISyntaxFacts syntaxFacts)
+        {
+            using var _ = ArrayBuilder<SnippetPlaceholder>.GetInstance(out var arrayBuilder);
+            var placeHolderBuilder = new Dictionary<string, List<int>>();
+            GetPartsOfForStatement(node, out var declaration, out var condition, out var incrementor, out var _1);
+
+            // Retrieves the placeholder present in the variable declaration as well as its location.
+            if (declaration != null)
```

Comment:
when is teh decl, condition, or incrementor null?  don't you always create all 3?